lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -101,6 +102,22 @@ llvm::Error setTensorType(const caffe2::QTensorProto &in, Tensor *T) {
     RETURN_ERR("Only uint8 and int32 qtensors are supported");
   }
 }
+
+Storage *createPlaceHolderFromWeightTensor(
+    PlaceholderBindings *bindings, Function &f, Tensor *weights,
+    const std::string &name, bool moveOrCopy, bool isQuantized,
+    std::function<void(Tensor *)> &&initializer,
+    ElemKind type = ElemKind::FloatTy, bool isTrainable = true) {
+  auto *ph = isQuantized
```

Comment:
Why do you need to create the Placeholder differently based on whether it's quantized or not? If you call `Placeholder *Module::createPlaceholder(TypeRef T, llvm::StringRef name, bool isTrainable)` with `&weights->getType()` as the parameter `T`, shouldn't the quantization information be encoded in the type?

Reply:
Yes, createPlaceholder -> uniqueType(...) must receive scale and offset explicitly.